src/test/sql/sql_translator_test.cpp

```diff
@@ -1996,4 +1996,91 @@ TEST_F(SQLTranslatorTest, CatchInputErrors) {
   EXPECT_THROW(compile_query("UPDATE no_such_table SET a = 1 WHERE a = 1"), InvalidInputException);
 }
 
+TEST_F(SQLTranslatorTest, WithClauseSingleQuerySimple) {
+  const auto actual_lqp = compile_query(
+      "WITH "
+      "with_query AS (SELECT a, b FROM int_int_int) "
+      "SELECT * FROM with_query WHERE a > 123;");
+  // clang-format off
+  const auto expected_lqp =
+      PredicateNode::make(greater_than_(int_int_int_a, value_(123)),
+          ProjectionNode::make(expression_vector(int_int_int_a, int_int_int_b),
+              stored_table_node_int_int_int));
+  // clang-format on
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
+TEST_F(SQLTranslatorTest, WithClauseSingleQueryAlias) {
+  const auto actual_lqp_alias = compile_query(
+      "WITH "
+      "with_query AS (SELECT a AS x FROM int_int_int) "
+      "SELECT * FROM with_query WHERE x > 123;");
```

Comment:
`SELECT * FROM with_query WHERE x > 123` would throw an exception, currently, right? I think this should be fixed.

Reply:
I do not quite understand.. Do you mean the `;` ?

Reply:
Can't reproduce what was meant either. Maybe he was thinking about `a > 123`, but that is addressed in a different issue.

Reply:
He probably meant `a >123` ;)